sdk/eventhub/azure-eventhub/migration_guide.md

```diff
@@ -1,266 +1,178 @@
+
+In v1:
 ```python
-client = EventHubClientAsync.from_connection_string(connection_str, eventhub=EVENTHUB_NAME)
-sender = client.add_async_sender(partition="0")
-try:
-    await client.run_async()
-    event_data = EventData(b"A single event")
-    await sender.send(event_data)
-finally:
-    await client.stop_async()
+    client = EventHubClient(address)
+    receiver = client.add_receiver(consumer_group, partition)
+    client.run()
+    batch = receiver.receive()
+    client.stop()
 ```
 
-In V5:
+In v5:
 ```python
-producer = EventHubProducerClient.from_connection_string(conn_str=EVENT_HUB_CONNECTION_STR, eventhub_name=EVENTHUB_NAME)
-async with producer:
-    event_data_batch = await producer.create_batch(partition_id="0")
-    event_data_batch.add(EventData(b"A single event"))
-    await producer.send_batch(event_data_batch)
+    # Receive
+    def on_event(partition_context, event):
+        print("Received event from partition: {}.".format(partition_context.partition_id))
+    
+    consumer_client = EventHubConsumerClient.from_connection_string(conn_str, consumer_group, eventhub_name=eh_name)
+    with consumer_client:
+        consumer_client.receive(on_event=on_event)
+
+    # Receive batch
+    def on_event_batch(partition_context, event_batch):
+        print("Partition {}, Received count: {}".format(partition_context.partition_id, len(event_batch)))
+        
+    consumer_client = EventHubConsumerClient.from_connection_string(conn_str, consumer_group, eventhub_name=eh_name)
+    with consumer_client:
+        consumer_client.receive_batch(on_event_batch=on_event_batch)
 ```
 
-### Migrating code from `EventProcessorHost` to `EventHubConsumerClient` for receiving events
-
-In V1, `EventProcessorHost` allowed you to balance the load between multiple instances of 
-your program when receiving events.
-
-In V5, `EventHubConsumerClient` allows you to do the same with the `receive()` method if you
-pass a `CheckpointStore` to the constructor.
-
-> **Note:** V1 checkpoints are not compatible with V5 checkpoints.
-If pointed at the same blob, consumption will begin at the first message.
-V1 checkpoint json in the respective blobs can be manually converted (per-partition) if needed.
-In V1 checkpoints (sequence_number and offset) are stored in the format of json along with ownership information
-as the content of the blob, while in V5, checkpoints are kept in the metadata of a blob and the metadata is composed of name-value pairs.
-Please check [update_checkpoint](https://github.com/Azure/azure-sdk-for-python/blob/master/sdk/eventhub/azure-eventhub-checkpointstoreblob/azure/eventhub/extensions/checkpointstoreblob/_blobstoragecs.py#L231-L250) in V5 for implementation detail.
+### Receiving with checkpoints
+Consuming events and saving checkpoints using a checkpoint store was not available in v1.
```

Comment:
Well... not true. We had the event processor host that allowed this.

Reply:
adding the ### Migrating code from `EventProcessorHost` to `EventHubConsumerClient` for receiving events
section back in